### PR TITLE
SO-3231: Update existing issues upon validation

### PR DESCRIPTION
**Pull request by norban**

### CODE REVIEW

File: core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java [hunk 1]
```diff
@@ -137,7 +140,11 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 								.execute(context)
 								.getItems();
 						
+						
 						final Set<ComponentIdentifier> existingComponentIdentifiers = existingRuleIssues.stream().map(ValidationIssue::getAffectedComponent).collect(Collectors.toSet());
+						
+						final Map<ComponentIdentifier, ValidationIssue> existingIsssuesByComponentIdentifier = existingRuleIssues.stream().collect(Collectors.toMap(ValidationIssue::getAffectedComponent, Function.identity()));
```
Comment (cmark) [inline core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java]: If you create an `affectedComponent` to `issue` `Map`, then you can safely remove the `existingComponentIdentifiers` variable and use the `keySet` of the `Map`.

**Review by cmark — APPROVED** [review]
LGTM!

**Comment by norban** [comment]
Requested change have been applied

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**